times_reader/transforms.py

```diff
@@ -353,50 +351,52 @@ def remove_comment_rows(table: datatypes.EmbeddedXlTable) -> datatypes.EmbeddedX
         return table
 
     df = table.dataframe.copy()
-    comment_rows = list(
-        locate(
-            df.iloc[:, 0],
-            lambda cell: isinstance(cell, str)
-            and (cell.startswith("*") or cell.startswith("\\I:")),
-        )
-    )
-    df.drop(index=comment_rows, inplace=True)
-    df.reset_index(drop=True, inplace=True)
+    colname = df.columns[0]
+
+    tag = table.tag.split(":")[0]
+
+    if tag in config.row_comment_chars:
+        chars_by_colname = config.row_comment_chars[tag]
+    else:
+        return table
 
-    # TODO: the deletion of this third column is a bug. Removing it causes the
-    # program to fail parse all rows. We need to fix the parsing so it can read
-    # all rows and remove this code block.
-    if df.shape[1] > 1:
-        comment_rows = list(
-            locate(
-                df.iloc[:, 1],
-                lambda cell: isinstance(cell, str) and cell.startswith("*"),
+    for colname in df.columns:
+        if colname in chars_by_colname.keys():
+            comment_rows = list(
+                locate(
+                    df[colname],
+                    lambda cell: isinstance(cell, str)
+                    and (cell.startswith(tuple(chars_by_colname[colname]))),
+                )
             )
-        )
-        df.drop(index=comment_rows, inplace=True)
-        df.reset_index(drop=True, inplace=True)
+            df.drop(index=comment_rows, inplace=True)
+            df.reset_index(drop=True, inplace=True)
+
     return replace(table, dataframe=df)
 
 
 def remove_comment_cols(table: datatypes.EmbeddedXlTable) -> datatypes.EmbeddedXlTable:
     """
     Return a modified copy of 'table' where columns with labels starting with '*'
-    have been deleted.
+    have been deleted. Assumes that any leading spaces in the original input table
+    have been removed.
 
     :param table:       Table object in EmbeddedXlTable format.
     :return:            Table object in EmbeddedXlTable format without comment columns.
     """
     if table.dataframe.size == 0:
         return table
 
-    comment_cols = list(
-        locate(
-            table.dataframe.columns,
-            lambda cell: isinstance(cell, str) and cell.startswith("*"),
-        )
-    )
-    df = table.dataframe.drop(table.dataframe.columns[comment_cols], axis=1)
+    comment_cols = [
+        colname
+        for colname in table.dataframe.columns
+        if isinstance(colname, str) and colname.startswith("*")
+    ]
+
+    df = table.dataframe.drop(comment_cols, axis=1)
     df.reset_index(drop=True, inplace=True)
+
+    # TODO: should we move the code below to a separate transform?
```

Comment:
No, Veda does not allow duplicate columns. This check/warning just seems out-of-place, because removal of comment columns won't cause duplicate columns.

Reply:
Good point. Perhaps we have a transform called `validate_input_tables` or something that checks that input tables are valid, i.e. no duplicates, valid tag names, etc?

Reply:
I like the idea. We could also use it to discard empty tables (instead of `remove_empty_tables`) and tables we won't process (i.e. see my comment in #138)